src/unix/fs.c

```diff
@@ -406,6 +396,67 @@ static ssize_t uv__fs_open(uv_fs_t* req) {
 }
 
 
+static ssize_t uv__preadv_emul(int fd,
+                               const struct iovec* bufs,
+                               size_t nbufs,
+                               off_t off) {
+  ssize_t nread;
+  ssize_t r;
+  size_t i;
+  size_t n;
+  void* p;
+
+  nread = 0;
+  for (i = 0; i < nbufs; i++) {
+    p = bufs[i].iov_base;
+    n = bufs[i].iov_len;
+
+    do
+      r = pread(fd, p, n, off);
+    while (r == -1 && errno == EINTR);
```

Comment:
The old code had this too and I think it should be fine because a) it's only on EINTR (doesn't affect offset), and b) pread/preadv are only used when off != -1. That pretty much excludes most /dev files and pipes.

Reply:
Ah, indeed that is right, I was thinking of `pread`. This is the case where we can do everything in parallel without issues.

Reply:
For others reference, the prior code was 663d88b677f22b74ed3002c1d704f87188692dc5